### ISSUE TITLE
 Unable to install through BRAT?

### ISSUE THREAD

**Issue by kickTiger** [willasm/vc-weather#1]
Unable to install through BRAT?

**Comment by willasm** [comment]
I actually have never used the Brat plugin myself but have the Github page open right now and will look into it. I am currently on Windows right now and I will need to boot to Linux where I have my development environment setup. I will install Brat then and test it for myself. Can you please tell me if it is giving you any error messages at all. Like I said I have never used it myself so I have no idea of what to expect when it has an error.

You can always install it manually if you need to. That involves creating the folder `visual-crossing-weather` inside your vaults `.obsidian\plugins` folder, then move the files `manifest.json` and `main.js` inside the newly created folder. You will need to either clone this repo to get the files or use another method to download them (there is a website that does that but I don't have the link on this system so you will need to search for it).

I'm going to switch to my development system now so I can look into this, William.

**Comment by willasm** [comment]
Ok I think I figured this out. It actually needs a release, it does not just grab the files directly. You will need to use the command with Frozen version based on a release tag, add the repo url and use the tag 0.0.1 and it will install it. Not sure it does it that way? Seems like it would be much easier just to download manifest file, extract the version from that and if it is a newer version then download main.json. Oh well, anyhow it does work now although I could not get it to work with the command that doesn't require the tag?

I assume you already have an API key. It wont work at all without it.

I will play around with Brat to see if I can get it to work with the command that doesn't require the tag.

**Comment by arcticpinecone** [comment]
Nice, came here just to find what I needed. 
Will this show up in a new Obsidian plugin? or?

**Comment by willasm** [comment]
I have it setup correctly for Brat now and you no longer need the tag to download the plugin. I have pinned the instructions for Brat & Manual installs at the top of the issues page.

@arcticpinecone This will install the plugin and it will show up in the settings community plugins list (you may need to enable it if it isn't already enabled). Note that you should disable the old OpenWeather plugin or they will conflict with each other.

**Comment by arcticpinecone** [comment]
> I have it setup correctly for Brat now and you no longer need the tag to download the plugin. I have pinned the instructions for Brat & Manual installs at the top of the issues page.

Noted, and thanks!

> @arcticpinecone This will install the plugin and it will show up in the settings community plugins list (you may need to enable it if it isn't already enabled). Note that you should disable the old OpenWeather plugin or they will conflict with each other.

I got it working, thanks! What I was looking for was more like, is this only going to be through BRAT now, or will I be able to subscribe to it like I used to (within Obsidian), like we had with the previous plugin?

![image](https://github.com/user-attachments/assets/be5be787-ad1e-429b-a6df-016c3786bda6)

**Comment by willasm** [comment]
Oh I see, it will be available from within Obsidian after it is published. I will be submitting it soon and after it is approved it will be added to the community plugins list. Until then the only way to install it is with Brat or manually.

Hope that clears that up for you, William

**Comment by kickTiger** [comment]
Thank you very much for your reply, 

I downloaded and installed the plugin and started it successfully, but I had trouble registering an account.

I registered on www.visualcrossing.com website, unfortunately my Gmail was flagged and blocked by him.

In fact, this Gmail has been used normally for a long time. I suspect it may be because some areas are blocked.

Later, I will change the region and email address to try to register to see if there is any restriction.

The following is the error message returned by registration.
```
There was a problem creating the account.
Your activity has been flagged for possible usage violations (1).
```

**Comment by willasm** [comment]
That is strange? Did you have an account with them before? From that message 'possible usage violations', that is what it sounds like? Might be blocked on a region of ip addresses maybe? I registered using my gmail address without any issues. Yes I would try another email address. If you are still having issues after that I would contact them, either through email or on their forum and ask them what is causing the problem.

I did figure out what Brat needed. I was marking the releases as Betas's which caused Brat to not download it. That is all fixed now.

I just uploaded another release just minutes ago. Added a modal that displays the current weather information. It can be activated by clicking anywhere on the statusbar weather info.

At this point I will not be adding any more features. All the ones from the old plugin are there and working as well as a couple new ones. I will now focus on the documentation. At the very least I need a Readme.md that covers all the basic details well enough to enable me to submit the plugin to Obsidian for reveiw. I hope to have it submited by the end of the weekend at the latest, barring finding any bugs in the meantime.

Thanks, William.
